### PR TITLE
Replace util.js polyfills with JS natives...

### PR DESCRIPTION
**Pull request by fredcw**
Replace all instances of `js/misc/util.js` polyfills (`findIndex(), find(), each(), filter() & map()`) with JS natives in cinnamon and delete these polyfills from `util.js`.

These were originally added for performance but are now slower in current cjs. They are no longer used by any xlets in linuxmint/cinnamon-spices-applets, linuxmint/cinnamon-spices-desklets, or linuxmint/cinnamon-spices-extensions. However, they are still used in the cinnamon menu applet so this PR depends on https://github.com/linuxmint/cinnamon/pull/11865 also being merged.

+grouped-window-list applet: Always add alert style to thumbnail when app window needs attention and implement same style removal when app is focused (not previously implemented)

+grouped-window-list applet: used `const` instead of `let` where possible.

### CODE REVIEW

File: js/misc/util.js
Comment (anaximeno) [inline js/misc/util.js:1]: This might break extensions that depend on these utils.

File: js/misc/util.js [hunk 1]
```diff
@@ -459,102 +459,6 @@ function queryCollection(collection, query, indexOnly = false) {
...
-    let len = arr.length;
-    let out = Array(len);
-
-    for (let i = 0; i < len; i++) {
-        out[i] = callback(arr[i], i, arr);
-    }
-
-    return out;
-};
-
```
Comment (anaximeno) [inline js/misc/util.js:557]: This might break extensions that depend on these utils.

**Comment by brownsr** [comment]
Applet fails to launch:

[grouped-window-list@cinnamon.org]: find is not a function
[grouped-window-list@cinnamon.org]: Failed to evaluate 'main' function on applet: grouped-window-list@cinnamon.org/16
Cjs-Message: 08:11:25.405: JS LOG: [LookingGlass/trace] 
<----------------
_connect@/usr/share/cinnamon/js/misc/state.js:250:24
connect@/usr/share/cinnamon/js/misc/state.js:271:25
GroupedWindowListApplet@/usr/share/cinnamon/applets/grouped-window-list@cinnamon.org/applet.js:211:20
main@/usr/share/cinnamon/applets/grouped-window-list@cinnamon.org/applet.js:1022:12
createApplet@/usr/share/cinnamon/js/ui/appletManager.js:597:25
addAppletToPanels@/usr/share/cinnamon/js/ui/appletManager.js:372:34
finishExtensionLoad@/usr/share/cinnamon/js/ui/appletManager.js:98:14
_init/<@/usr/share/cinnamon/js/ui/extension.js:275:32
---------------->
Cjs-Message: 08:11:25.406: JS LOG: [LookingGlass/error] 
[grouped-window-list@cinnamon.org]: Applet grouped-window-list@cinnamon.org: Could not create applet object.
[grouped-window-list@cinnamon.org]: Error importing applet.js from grouped-window-list@cinnamon.org
Cjs-Message: 08:11:25.406: JS LOG: [LookingGlass/trace]

**Comment by fredcw** [comment]
@brownsr Thanks. Yes, I changed state.js and then forgot to add it to the commit 🤦‍♂️. I'll add it in a new PR

**Comment by fredcw** [comment]
@mtwebster Unrelated to this PR but I was wondering: in GWL applet.js are the following comments:

line 205
```
// key-function pairs of actions that can be triggered from the store's callback queue. This allows the
// applet to avoid passing down the parent class down the constructor chain and creating circular references.

```
line 262
```
// Passing an empty object instead of `this` because its only used by SignalManager to bind the callback, which
// we already do here. Otherwise, it creates more circular references.
```

line 845
```
// TODO: Figure out exactly which properties on this applet constructor the Cinnamon APIs needs for all modes of
// DND, so we can kill the _delegate reference. Long term, a PR to Cinnamon should be opened fixing circular
// object reference structures for the applet and desklet classes.
```

Do you know of any reason why circular references should be avoided or perhaps this isn't necessary now?

**Comment by mtwebster** [comment]
Seeing as the gwl applet is really the only place we worry about any of this (in Cinnamon proper, at least), I doubt this is as much of an issue as was being made here. Some googling also gives me the impression JS garbage collection is robust enough to handle it np.

We end up with circular refs all over the place now using _delegate in popup menus, xlets, etc... so even if it *is* an issue with GC it doesn't seem like a major one. We could always clear _delegates when we're cleaning up these things - that seems like it'd be easier than tying our hands ahead of time. That second snippet just seems like it'd be confusing.

**Pull request merged**